src/main/java/Bart.java

```diff
@@ -0,0 +1,86 @@
+import java.util.Scanner;
+
+public class Bart {
+    private static final String LINE = "____________________________________________________________";
+    private static final Task[] tasksArray = new Task[100];
+    private static int taskCount = 0;
+
+    public static void main(String[] args) {
+        greetUser();
+        manageTask();
+        byeUser();
+    }
+    public static void greetUser() {
+        System.out.println(LINE);
+        System.out.println("Hello! I'm Bartholomew, but you can call me Bart for short :)");
+        System.out.println("What can I do for you?");
+        System.out.println(LINE);
+    }
+    public static void manageTask() {
+        Scanner in = new Scanner(System.in);
+        String command = "";
+        while (!command.equals("bye")) {
+            command = in.nextLine();
+            if (command.equals("list")) {
+                listTasks();
+            } else if (command.startsWith("mark")) {
+                markTask(command);
+            } else if (command.startsWith("unmark")) {
+                unmarkTask(command);
+            } else {
+                if (taskCount < tasksArray.length) {
+                    tasksArray[taskCount] = new Task(command); // Create new Task object
+                    taskCount++;
+                    System.out.println(LINE);
+                    System.out.println("added: " + command);
+                    System.out.println(LINE);
+                } else {
+                    System.out.println("Sorry, task list is full.");
+                }
+            }
+        }
+    }
+    public static void listTasks() {
+        System.out.println(LINE);
+        //Edge case: If list empty
+        if (taskCount == 0) {
+            System.out.println("Nothing added");
+        }
+        for (int i = 0; i < taskCount; i++) {
+            System.out.println((i + 1) + "." + tasksArray[i].getStatusIcon() + " " + tasksArray[i].description);
+        }
+        System.out.println(LINE);
+    }
+
+    public static void markTask(String command) {
+        int taskIndex = Integer.parseInt(command.substring(5).trim()) - 1;
+
+        // Check if the task index is within the valid range
+        if (taskIndex >= 0 && taskIndex < taskCount) {
+            tasksArray[taskIndex].markAsDone();
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(tasksArray[taskIndex].getStatusIcon() + " " + tasksArray[taskIndex].description);
+        } else {
+            System.out.println("Invalid task number.");
+        }
+        System.out.println(LINE);
+    }
+
+    //similar to markTask
+    public static void unmarkTask(String command) {
+        int taskIndex = Integer.parseInt(command.substring(7).trim()) - 1;
```

Comment:
.substring(7) seems to be hard-coded, and assumes the user input to follow a certain format. You could consider making the starting index dynamic, while providing some visual guide for the user, like in the Contacts example we did in class.